### PR TITLE
Resources Endpoints

### PR DESCRIPTION
**Pull request by dennisroberts71**
This change adds CRUD operations for resources in the permissions database.

### CODE REVIEW

File: golang/src/permissions/restapi/impl/db/resource_types.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func GetDuplicateResourceTypeByName(tx *sql.Tx, id *string, name *string) (*mode
 	defer rows.Close()
 
 	// Get the resource type.
-	var resourceTypes []*models.ResourceTypeOut
+	resourceTypes := make([]*models.ResourceTypeOut, 0)
```
Comment (ianmcorvidae) [inline golang/src/permissions/restapi/impl/db/resource_types.go]: I have no doubt this is correct but for the sake of my own education, why this change?

**Comment by ianmcorvidae** [comment]
LGTM. Seems like pretty straightforward CRUD stuff so as long as there's integration tests seems ready to ship.

File: golang/src/permissions/restapi/impl/db/resource_types.go (hunk 1 above)
Comment (dennisroberts71) [inline golang/src/permissions/restapi/impl/db/resource_types.go]: In the original statement, `resourceTypes` was initialized to `nil`. Normally, this wasn't a problem because appending to a nil slice produces a single-element slice. If there were no list items, however, then the `nil` was being translated to `null` in the JSON that is produced by the endpoint. Using `make` produces an actual empty slice so that the JSON will contain an empty array instead.

Comment (ianmcorvidae) [inline golang/src/permissions/restapi/impl/db/resource_types.go]: Ah, cool, that makes sense. Thanks!
